src/lib/hooks/useEnableKeyboardController.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react'
+import {KeyboardProvider} from 'react-native-keyboard-controller'
+import {useFocusEffect} from '@react-navigation/native'
+
+const KeyboardControllerRefCountContext = createContext<{
+  incrementRefCount: () => void
+  decrementRefCount: () => void
+}>({
+  incrementRefCount: () => {},
+  decrementRefCount: () => {},
+})
+
+export function KeyboardControllerProvider({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  const [refCount, setRef] = useState(0)
+
+  const value = useMemo(
+    () => ({
+      incrementRefCount: () => setRef(count => count + 1),
+      decrementRefCount: () => setRef(count => Math.max(count - 1, 0)),
```

Comment:
as long as we're doing them correctly, it should never dip below 0 so i would remove this check. instead, could console.error in render if it's negative